core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/http/pipeline/stages/SigningStage.java

```diff
@@ -84,16 +84,11 @@ private <T extends Identity> SdkHttpFullRequest sraSignRequest(SdkHttpFullReques
             () -> {
                 AuthSchemeOption authSchemeOption = selectedAuthScheme.authSchemeOption();
 
-                // TODO: Identity resolution should move to before Endpoint resolution interceptor, to support accountId based
-                //  endpoints and also to logically separate out identity resolution as its own step.
-                ResolveIdentityRequest.Builder identityRequestBuilder = ResolveIdentityRequest.builder();
-                authSchemeOption.forEachIdentityProperty(identityRequestBuilder::putProperty);
-
-                IdentityProvider<T> identityProvider = selectedAuthScheme.identityProvider();
-                T identity = identityProvider.resolveIdentity(identityRequestBuilder.build()).join();
+                CompletableFuture<T> identityFuture =
+                    (CompletableFuture<T>) executionAttributes.getAttribute(SdkInternalExecutionAttribute.IDENTITY);
 
                 SyncSignRequest.Builder<T> signRequestBuilder = SyncSignRequest
-                    .builder(identity)
+                    .builder(CompletableFutureUtils.joinLikeSync(identityFuture))
```

Comment:
is this fine in SigningStage? AsyncSigningStage would do async logic.

Reply:
yes